### PR TITLE
chore: Update dependencies

### LINKED ISSUE #2168: Update outdated dependencies
The dependencies for this repository are out of date. Please check the list here: https://david-dm.org/fossasia/open-event-wsgen

Update the dependencies.

### PR DESCRIPTION
**Pull request by djcruz93**
#### Checklist

- [x]  I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Update all the outdated dependencies referring to https://david-dm.org/fossasia/open-event-wsgen. The update resulted in some breaking changes which required the codebase to change. 

#### Changes proposed in this pull request:
- Update the following dependencies:
  - del v6.0.0
  - folder-hash v4.0.0
  - fs-extra v9.0.1
  - github v14.0.0
  - gulp-babel v8.0.0
- Latest version of fs-extra sets err variable as undefined instead of null so removed redundant if condition checks of err with null
- Add and set preset @babel/preset-env
- gulp-babel has a peer dependency on babel/core v7.0.0
- ftp-deploy has peer dependency on promise-ftp-common v1.1.5
- github v14.0.0 has been renamed to octokit/rest v14.0.0
- Change the module "github" to "@octokit/rest" in deploy.js while importing

Fixes #2168

### CODE REVIEW

**Comment by djcruz93** [comment]
@iamareebjamal this won't run the travis ci build as I have changed only the package.json. I have checked there is no build corresponding to this PR as well.

**Comment by iamareebjamal** [comment]
You need to update yarn.lock as well and travis build should still work

File: src/backend/deploy.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 const distHelper = require('./dist.js');
 const fs = require('fs');
-const Github = require('github');
+const Github = require('@octokit/rest');
```
Comment (iamareebjamal) [inline src/backend/deploy.js:6]: Nothing else needed to be changed?

Comment (djcruz93) [reply src/backend/deploy.js:6]: No

**Pull request merged**